### PR TITLE
Feature/data ecs instances

### PR DESCRIPTION
**Pull request by gllb**

### CODE REVIEW

File: docs/data-sources/compute_instances.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+subcategory: "Elastic Cloud Server (ECS)"
+---
+
+# flexibleengine\_compute\_instances
```
Comment (ShiChangkuo) [inline docs/data-sources/compute_instances.md:5]: - **\\** is unnecessary
- should we add "_v2" suffix to keep consistent with other resources and data sources?

File: docs/data-sources/compute_instances.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Data source ID.
+
+* `instances` - List of ECS instance details. The object structure of each ECS instance is documented below.
+
+The `instances` block supports:
+
+* `id` - The instance ID in UUID format.
```
Comment (ShiChangkuo) [inline docs/data-sources/compute_instances.md:54]: `name` is missing

File: flexibleengine/data_source_flexibleengine_compute_instances_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("Compute instance data source ID not set")
+		}
+
+		return nil
+	}
+}
+
+func testAccComputeInstancesDataSource_basic(rName string) string {
```
Comment (ShiChangkuo) [inline flexibleengine/data_source_flexibleengine_compute_instances_test.go:58]: `rName` is defined but not ued

File: flexibleengine/data_source_flexibleengine_compute_instances_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	return fmt.Sprintf(`
+%s
+resource "flexibleengine_compute_instance_v2" "instance_2" {
+  name               = "instance_2"
+  security_groups = ["default"]
+  availability_zone  = "%s"
```
Comment (ShiChangkuo) [inline flexibleengine/data_source_flexibleengine_compute_instances_test.go:64]: When multiple arguments with single-line values appear on consecutive lines at the same nesting level,  please align their equals signs.

File: flexibleengine/data_source_flexibleengine_compute_instances.go [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+
+	for i, val := range servers {
+		server := map[string]interface{}{
+			"id":                val.ID,
+			"user_data":         val.UserData,
+			"name":              val.Name,
+			"flavor_name":       val.Flavor.Name,
+			"status":            val.Status,
+			"flavor_id":         val.Flavor.ID,
+			"image_id":          val.Image.ID,
```
Comment (ShiChangkuo) [inline flexibleengine/data_source_flexibleengine_compute_instances.go:246]: **image_name** is missing

File: flexibleengine/data_source_flexibleengine_compute_instances.go [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+						},
+						"security_groups": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"system_disk_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
```
Comment (ShiChangkuo) [inline flexibleengine/data_source_flexibleengine_compute_instances.go:108]: `system_disk_id` is defined but not set

File: docs/data-sources/compute_instances.md (hunk 1 above)
Comment (gllb) [reply docs/data-sources/compute_instances.md]: "\" is removed.
In my opinion Terraform abstract the API call so a resource should not contain the version of the API Terraform is calling. And a change in the API should result in a new version of the terraform provider so user can select the appropriate version for their usage. What do you think about it ?

File: docs/data-sources/compute_instances.md (hunk 2 above)
Comment (gllb) [reply docs/data-sources/compute_instances.md:54]: done

File: flexibleengine/data_source_flexibleengine_compute_instances_test.go (hunk 3 above)
Comment (gllb) [reply flexibleengine/data_source_flexibleengine_compute_instances_test.go]: done

File: flexibleengine/data_source_flexibleengine_compute_instances_test.go (hunk 4 above)
Comment (gllb) [reply flexibleengine/data_source_flexibleengine_compute_instances_test.go]: done

File: flexibleengine/data_source_flexibleengine_compute_instances.go (hunk 5 above)
Comment (gllb) [reply flexibleengine/data_source_flexibleengine_compute_instances.go:242]: done

File: flexibleengine/data_source_flexibleengine_compute_instances.go (hunk 6 above)
Comment (gllb) [reply flexibleengine/data_source_flexibleengine_compute_instances.go]: done

File: docs/data-sources/compute_instances.md (hunk 1 above)
Comment (ShiChangkuo) [reply docs/data-sources/compute_instances.md]: yes, we can add new resource or data source without API version.

File: flexibleengine/data_source_flexibleengine_compute_instances.go (hunk 5 above)
Comment (ShiChangkuo) [reply flexibleengine/data_source_flexibleengine_compute_instances.go:242]: you can get the attribute from server.Metadata

**Comment by ShiChangkuo** [comment]
@gllb Thanks for your efforts. I suggest to pack all attributes of [flexibleengine_compute_instance](https://registry.terraform.io/providers/FlexibleEngineCloud/flexibleengine/latest/docs/data-sources/compute_instance_v2) into the object of  `instances` block.

**Comment by gllb** [comment]
@ShiChangkuo, yes I packed all attributes but the `block_device` as the API do not give enough information. If we want them we need to make an extra API call for each volume. I thought it was better to have a dedicated volume data_source that could be used for that purpose and the `volume_attached` attributes give enough information to retrieve the volume by ID. It´s inspired from https://github.com/huaweicloud/terraform-provider-huaweicloud/blob/master/huaweicloud/data_source_huaweicloud_compute_instances.go#L109-L124

**Comment by gllb** [comment]
```
$ make testacc TEST='./flexibleengine' TESTARGS='-run TestAccComputeInstance*'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./flexibleengine -v -run TestAccComputeInstance* -timeout 720m
=== RUN   TestAccComputeInstanceDataSource_basic
=== PAUSE TestAccComputeInstanceDataSource_basic
=== RUN   TestAccComputeInstancesDataSource_basic
=== PAUSE TestAccComputeInstancesDataSource_basic
=== CONT  TestAccComputeInstanceDataSource_basic
--- PASS: TestAccComputeInstanceDataSource_basic (57.58s)
=== CONT  TestAccComputeInstancesDataSource_basic
--- PASS: TestAccComputeInstancesDataSource_basic (47.38s)
PASS
ok      github.com/terraform-providers/terraform-provider-flexibleengine/flexibleengine 104.982s

```

File: flexibleengine/data_source_flexibleengine_compute_instances.go (hunk 5 above)
Comment (gllb) [reply flexibleengine/data_source_flexibleengine_compute_instances.go:242]: thanks, it´s done

File: docs/data-sources/compute_instances.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+* `name` - (Optional) Specifies the server name, which can be queried with a regular expression.
```
Comment (ShiChangkuo) [inline docs/data-sources/compute_instances.md:56]: suggest writing:
```
* `name` - The instance name.
```

Comment (gllb) [reply docs/data-sources/compute_instances.md]: done

**Pull request merged**